File: Code/New folder/Mobile/lib/models/notification_preference.dart
class NotificationPreference {
  final String preferenceID;
  final String userID;
  final Map<String, dynamic> preferences;
  final DateTime createdAt;
  final DateTime updatedAt;

  NotificationPreference({
    required this.preferenceID,
    required this.userID,
    required this.preferences,
    required this.createdAt,
    required this.updatedAt,
  });

  factory NotificationPreference.fromJson(Map<String, dynamic> json) {
    return NotificationPreference(
      preferenceID: json['preferenceID'] as String,
      userID: json['userID'] as String,
      preferences: json['preferences'] as Map<String, dynamic>,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'preferenceID': preferenceID,
      'userID': userID,
      'preferences': preferences,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}

File: Code/New folder/Back/models/notif/notificationPreference.js
const { nanoid } = require('nanoid');

module.exports = (sequelize, DataTypes) => {
    const NotificationPreference = sequelize.define('NotificationPreference', {
        preferenceID: {
            type: DataTypes.STRING,
            primaryKey: true,
            defaultValue: () => `pref_${nanoid()}`,
            allowNull: false,
        },
        userID: {
            type: DataTypes.STRING,
            allowNull: false,
            unique: true,
            references: {
                model: 'Users',
                key: 'userID',
            },
        },
        preferences: {
            type: DataTypes.JSON,
            allowNull: false,
            defaultValue: {},
        },
        createdAt: {
            type: DataTypes.DATE,
            allowNull: false,
            defaultValue: DataTypes.NOW,
        },
        updatedAt: {
            type: DataTypes.DATE,
            allowNull: false,
            defaultValue: DataTypes.NOW,
        },
    }, {
        tableName: 'NotificationPreferences',
        timestamps: true,
    });

    return NotificationPreference;
};